packages/amazon/src/serverGroup/configure/wizard/instanceType/InstanceTypeSelector.tsx

```diff
@@ -0,0 +1,142 @@
+import { SimpleModeSelector } from './simpleMode/SimpleModeSelector';
+
+export interface IInstanceTypeSelectorProps {
+  command: IAmazonServerGroupCommand;
+  setFieldValue: (field: keyof IAmazonServerGroupCommand, value: any, shouldValidate?: boolean) => void;
+  instanceTypeDetails: IInstanceTypeCategory[];
+}
+
+export function InstanceTypeSelector(props: IInstanceTypeSelectorProps) {
+  const isLaunchTemplatesEnabled = AWSProviderSettings.serverGroups?.enableLaunchTemplates;
+
+  const [useSimpleMode, setUseSimpleMode] = React.useState(props.command.viewState.useSimpleInstanceTypeSelector);
+  const [unlimitedCpuCredits, setUnlimitedCpuCredits] = useState(props.command.unlimitedCpuCredits);
+
+  useEffect(() => {
+    if (props.command.unlimitedCpuCredits !== unlimitedCpuCredits) {
+      props.setFieldValue('unlimitedCpuCredits', unlimitedCpuCredits);
+    }
+  }, [unlimitedCpuCredits]);
+
+  const handleModeChange = (useSimpleModeNew: boolean) => {
+    if (useSimpleMode !== useSimpleModeNew) {
+      setUseSimpleMode(useSimpleModeNew);
+
+      // update viewState
+      props.setFieldValue('viewState', {
+        ...props.command.viewState,
+        useSimpleInstanceTypeSelector: useSimpleModeNew,
+      });
+
+      // update selected instance type(s) if mode changed.
+      // Simple mode uses command.instanceType to track selected type. Advanced mode uses command.launchTemplateOverridesForInstanceType to track selected types.
+      const multipleInstanceTypesInProps = props.command.launchTemplateOverridesForInstanceType;
+      const singleInstanceTypeInProps = props.command.instanceType;
+
+      if (useSimpleModeNew && multipleInstanceTypesInProps && multipleInstanceTypesInProps.length > 0) {
+        // advanced mode -> simple mode, pick the instance type with highest priority from command.launchTemplateOverridesForInstanceType
+        const highestPriorityNum = Math.min(...multipleInstanceTypesInProps.map((it) => it.priority));
+        const instanceTypeWithHighestPriority = multipleInstanceTypesInProps.find(
+          (it) => it.priority === highestPriorityNum,
+        ).instanceType;
+
+        props.setFieldValue('instanceType', instanceTypeWithHighestPriority);
+        props.setFieldValue('launchTemplateOverridesForInstanceType', []);
+        props.command.instanceTypeChanged(props.command);
+      } else if (!useSimpleModeNew && singleInstanceTypeInProps) {
+        // simple mode -> advanced mode, port command.instanceType to command.launchTemplateOverridesForInstanceType
+        const instanceTypes: IAmazonInstanceTypeOverride[] = [
+          {
+            instanceType: singleInstanceTypeInProps,
+            priority: 1,
+          },
+        ];
+        props.setFieldValue('launchTemplateOverridesForInstanceType', instanceTypes);
+        props.setFieldValue('instanceType', undefined);
+        props.command.instanceTypesChanged(props.command);
+      }
+    }
+  };
```

Comment:
```suggestion
      const isSimple = useSimpleModeNew && multipleInstanceTypesInProps?.length;
      const isAdvanced = !useSimpleModeNew && singleInstanceTypeInProps;
      let newInstanceType, newLTOverridesForInstanceType;
      
      if (isSimple) {
        const highestPriorityNum = Math.min(...multipleInstanceTypesInProps.map((it) => it.priority));
        const instanceTypeWithHighestPriority = multipleInstanceTypesInProps.find(
          (it) => it.priority === highestPriorityNum,
        ).instanceType;
        newInstanceType = instanceTypeWithHighestPriority;
        newLTOverridesForInstanceType = [];
      } 
      
      if (isAdvanced) {
        const instanceTypes: IAmazonInstanceTypeOverride[] = [{ instanceType: singleInstanceTypeInProps, 
        priority: 1 }];
        newLTOverridesForInstanceType = instanceTypes;
      }
    }
    
     props.setFieldValue('launchTemplateOverridesForInstanceType',  newLTOverridesForInstanceType);
     props.setFieldValue('instanceType', newInstanceType);
     props.command.instanceTypesChanged(props.command);
  };
```

Reply:
Thanks for the simplification. Extracting the conditionals into constants add clarity - will change that. 

The other changes where setFieldValue is called outside the if block isn't safe to do as the field values need to change only if the conditions are satisfied. 
To pull these out I will need to assign `newInstanceType` to `singleInstanceTypeInProps` first and then reassign conditionally - I think this adds more confusion than bring clarity. 

Also, there are 2 different calls made in if and else block - `command.instanceTypeChanged` and `command.instanceTypeSChanged` - I renamed the latter to `command.launchTemplateOverridesChanged`  for clarity.